packages/@aws-cdk/aws-servicediscovery/lib/private-dns-namespace.ts

```diff
@@ -0,0 +1,68 @@
+import ec2 = require('@aws-cdk/aws-ec2');
+import cdk = require('@aws-cdk/cdk');
+import { BaseNamespaceProps, NamespaceBase, NamespaceType } from './namespace';
+import { DnsServiceProps, Service } from './service';
+import { CfnPrivateDnsNamespace} from './servicediscovery.generated';
+
+export interface PrivateDnsNamespaceProps extends BaseNamespaceProps {
+  /**
+   * The Amazon VPC that you want to associate the namespace with.
+   */
+  vpc: ec2.IVpcNetwork;
+}
+
+/**
+ * Define a Service Discovery HTTP Namespace
+ */
+export class PrivateDnsNamespace extends NamespaceBase {
+  /**
+   * A name for the PrivateDnsNamespace.
+   */
+  public readonly namespaceName: string;
+
+  /**
+   * Namespace Id for the PrivateDnsNamespace.
+   */
+  public readonly namespaceId: string;
+
+  /**
+   * Namespace Arn for the namespace.
+   */
+  public readonly namespaceArn: string;
+
+  /**
+   * Type of the namespace.
+   */
+  public readonly type: NamespaceType;
+
+  /**
+   * The Amazon VPC that you want to associate the namespace with.
+   */
+  public readonly vpc: ec2.IVpcNetwork;
+
+  constructor(scope: cdk.Construct, id: string, props: PrivateDnsNamespaceProps) {
+    super(scope, id);
+    if (props.vpc === undefined) {
```

Comment:
That's impossible per the type definition of `PrivateDnsNamespaceProps`.

Reply:
True, but it doesn't hurt and might help with other language bindings.